cloudsync_onedrive.py

```diff
@@ -1137,10 +989,18 @@ def _make_path_relative_to_shared_folder_if_needed(self, path, force=False):
 
     def _info_from_rest(self, item, root=None):
         if not root:
-            raise NotImplementedError()
+            # parentReference.path format is: "drives/{drive_id}/root:/path/to/file"
+            # To get the file path, everything up to and including the first ":" can be ignored
+            # See: https://docs.microsoft.com/en-us/graph/api/resources/itemreference?view=graph-rest-1.0
+            # Note:
+            #   - if parentReference.path is missing -- item is the drive root dir
+            #   - if parentReference.path == "drives/{drive_id}/root:" -- parent is the drive root dir
+            parent_ref = item["parentReference"]
+            if "path" in parent_ref:
+                root = urllib.parse.unquote(parent_ref["path"].split(":", 1)[1]) or "/"
```

Comment:
do we have a test  for this new stuff?

Reply:
it is hit pretty hard everywhere, especially `test_rename` and `test_root_rename`.

I can add an explicit test though.